apps/wallet/src/ui/app/pages/site-connect/index.tsx

```diff
@@ -29,17 +32,18 @@ function SiteConnectPage() {
 	);
 	const dispatch = useAppDispatch();
 	const permissionRequest = useAppSelector(permissionSelector);
-	const activeAddress = useActiveAddress();
-	const [accountsToConnect, setAccountsToConnect] = useState<string[]>(() =>
-		activeAddress ? [activeAddress] : [],
+	const activeAccount = useActiveAccount();
+	const { data: accounts } = useAccounts();
+	const [accountsToConnect, setAccountsToConnect] = useState<SerializedUIAccount[]>(() =>
+		activeAccount ? [activeAccount] : [],
 	);
 	const handleOnSubmit = useCallback(
 		async (allowed: boolean) => {
 			if (requestID && accountsToConnect && permissionRequest) {
 				await dispatch(
 					respondToPermissionRequest({
 						id: requestID,
-						accounts: allowed ? accountsToConnect : [],
+						accounts: allowed ? accountsToConnect.map((account) => account.address) : [],
```

Comment:
We probably want to remove duplicate addresses here. (for now maybe later we can change the permissions as well to use account ids and not addresses)

Reply:
will follow-up in the unlock / lock accounts branch

Reply:
same here